service/lambda/api_op_AddLayerVersionPermission.go

```diff
@@ -208,6 +210,54 @@ func (m *opAddLayerVersionPermissionResolveEndpointMiddleware) HandleSerialize(c
 		)
 	}
 
+	authSchemes, err := internalauth.GetAuthenticationSchemes(&resolvedEndpoint.Properties)
+	if err != nil {
+		var nfe *internalauth.NoAuthenticationSchemesFoundError
+		if errors.As(err, &nfe) {
+			// if no auth scheme is found, default to sigv4
+			signingName := "lambda"
+			signingRegion := m.BuiltInResolver.(*BuiltInResolver).Region
+			ctx = awsmiddleware.SetSigningName(ctx, signingName)
+			ctx = awsmiddleware.SetSigningRegion(ctx, signingRegion)
+
+		}
+		authSchemes = []internalauth.AuthenticationScheme{}
+	}
+
+	for _, authScheme := range authSchemes {
+		switch v := authScheme.(type) {
+		case *internalauth.AuthenticationSchemeV4:
+			v4Scheme, _ := authScheme.(*internalauth.AuthenticationSchemeV4)
+			var signingName, signingRegion string
+			if v4Scheme.SigningName == nil {
+				signingName = "lambda"
+			}
+			if v4Scheme.SigningRegion == nil {
+				signingRegion = m.BuiltInResolver.(*BuiltInResolver).Region
+			}
+			ctx = awsmiddleware.SetSigningName(ctx, signingName)
+			ctx = awsmiddleware.SetSigningRegion(ctx, signingRegion)
+			break
+		case *internalauth.AuthenticationSchemeV4A:
+			v4aScheme, _ := authScheme.(*internalauth.AuthenticationSchemeV4A)
+			var signingName string
+			if v4aScheme.SigningName == nil {
+				signingName = "lambda"
+			}
+			ctx = awsmiddleware.SetSigningName(ctx, signingName)
+			ctx = awsmiddleware.SetSigningRegion(ctx, v4aScheme.SigningRegionSet[0])
+			break
+		case *internalauth.AuthenticationSchemeNone:
+			break
+		default:
```

Comment:
**correctness**: This will error out possibly before exhausting the set of auth schemes.

e.g. if the auth schemes returned looked like: `[notSupported, sigv4]` we would fail too soon. The default case should just be a `continue` and you'll have to keep track of whether a valid authscheme was found via another variable.

e.g.

```go
foundSupportedAuthScheme := false

for _, authScheme in range authSchemes {
      case supported1:
          foundSupportedAuthScheme = true
          ...
      case supported2:
          foundSupportedAuthScheme = true
          ...
}

if len(authSchemes) >= && !foundSupportedAuthScheme {
    return out, metadata, fmt.Errorf("....")
}
```

Reply:
doh yah thats right ... brain was a bit fried last night, will fix